tests/components/tradfri/test_config_flow.py

```diff
@@ -244,13 +254,34 @@ async def test_import_duplicate_aborted(hass):
 async def test_duplicate_discovery(hass, mock_auth, mock_entry_setup):
     """Test a duplicate discovery in progress is ignored."""
     result = await hass.config_entries.flow.async_init(
-        "tradfri", context={"source": "zeroconf"}, data={"host": "123.123.123.123"}
+        "tradfri",
+        context={"source": "homekit"},
+        data={"host": "123.123.123.123", "properties": {"id": "homekit-id"}},
     )
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
 
     result2 = await hass.config_entries.flow.async_init(
-        "tradfri", context={"source": "zeroconf"}, data={"host": "123.123.123.123"}
+        "tradfri",
+        context={"source": "homekit"},
+        data={"host": "123.123.123.123", "properties": {"id": "homekit-id"}},
     )
 
     assert result2["type"] == data_entry_flow.RESULT_TYPE_ABORT
+
+
+async def test_discovery_updates_unique_id(hass):
+    """Test a duplicate discovery host aborts and updates existing entry."""
+    entry = MockConfigEntry(domain="tradfri", data={"host": "some-host"},)
+    entry.add_to_hass(hass)
+
+    flow = await hass.config_entries.flow.async_init(
+        "tradfri",
+        context={"source": "homekit"},
+        data={"host": "some-host", "properties": {"id": "homekit-id"}},
+    )
+
+    assert flow["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert flow["reason"] == "already_configured"
+
+    assert entry.unique_id == "homekit-id"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5422434)